src/plugins/orangehrmAdminPlugin/Api/I18NTranslationBulkAPI.php

```diff
@@ -0,0 +1,157 @@
+ *
+ * OrangeHRM is free software; you can redistribute it and/or modify it under the terms of
+ * the GNU General Public License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Admin\Api;
+
+use Exception;
+use OrangeHRM\Admin\Traits\Service\LocalizationServiceTrait;
+use OrangeHRM\Core\Api\CommonParams;
+use OrangeHRM\Core\Api\V2\Endpoint;
+use OrangeHRM\Core\Api\V2\EndpointCollectionResult;
+use OrangeHRM\Core\Api\V2\EndpointResult;
+use OrangeHRM\Core\Api\V2\Exception\BadRequestException;
+use OrangeHRM\Core\Api\V2\Model\ArrayModel;
+use OrangeHRM\Core\Api\V2\ParameterBag;
+use OrangeHRM\Core\Api\V2\RequestParams;
+use OrangeHRM\Core\Api\V2\ResourceEndpoint;
+use OrangeHRM\Core\Api\V2\Validator\ParamRule;
+use OrangeHRM\Core\Api\V2\Validator\ParamRuleCollection;
+use OrangeHRM\Core\Api\V2\Validator\Rule;
+use OrangeHRM\Core\Api\V2\Validator\Rules;
+use OrangeHRM\Core\Traits\ORM\EntityManagerHelperTrait;
+use OrangeHRM\ORM\Exception\TransactionException;
+
+class I18NTranslationBulkAPI extends Endpoint implements ResourceEndpoint
+{
+    use EntityManagerHelperTrait;
+    use LocalizationServiceTrait;
+
+    public const PARAMETER_LANGUAGE_ID = 'languageId';
+    public const PARAMETER_DATA = 'data';
+    public const PARAMETER_TRANSLATED_VALUE = 'translatedValue';
+    public const PARAMETER_LANG_STRING_ID = 'langStringId';
+
+    /**
+     * @inheritDoc
+     */
+    public function getOne(): EndpointResult
+    {
+        throw $this->getNotImplementedException();
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForGetOne(): ParamRuleCollection
+    {
+        throw $this->getNotImplementedException();
+    }
+
+    /**
+     * @inheritDoc
+     * @throws BadRequestException
+     * @throws TransactionException
+     */
+    public function update(): EndpointCollectionResult
+    {
+        $this->beginTransaction();
+        try {
+            $languageId = $this->getRequestParams()->getInt(
+                RequestParams::PARAM_TYPE_ATTRIBUTE,
+                self::PARAMETER_LANGUAGE_ID
+            );
+
+            $translatedDataValues = $this->getRequestParams()->getArray(
+                RequestParams::PARAM_TYPE_BODY,
+                self::PARAMETER_DATA
+            );
+
+            $this->getLocalizationService()
+                ->saveAndUpdateTranslatedStringsFromRows($languageId, $translatedDataValues);
```

Comment:
```suggestion
            $this->getLocalizationService()
                ->saveAndUpdateTranslatedStringsFromRows($languageId, $translatedDataValues);
            $this->commitTransaction();
```
commit the transaction within try block